actionpack/test/controller/redirect_test.rb

```diff
@@ -335,6 +344,18 @@ def test_redirect_to_url_with_complex_scheme
     assert_equal "x-test+scheme.complex:redirect", redirect_to_url
   end
 
+  def test_redirect_to_url_with_relative_url_without_leading_slash
+    get :redirect_to_relative_url_without_leading_slash
+    assert_response :redirect
+    assert_equal "http://test.hostexmaple.com", redirect_to_url
```

Comment:
this is just documenting the unexpected status quo - not sure if we really should keep them